wcfsetup/install/files/js/WCF.Like.js

```diff
@@ -7,412 +7,25 @@
-				if (this._showSummary) this._updateSummary($containerID);
-				
-				// mark button as active
-				var $likeButton = this._containerData[$containerID].likeButton;
-				var $dislikeButton = this._containerData[$containerID].dislikeButton;
-				var $likeStatus = 0;
-				if (data.returnValues.isLiked) $likeStatus = 1;
-				else if (data.returnValues.isDisliked) $likeStatus = -1;
-				this._setActiveState($likeButton, $dislikeButton, $likeStatus);
-				
-				// invalidate cache for like details
-				if (this._likeDetails[$containerID] !== undefined) {
-					delete this._likeDetails[$containerID];
-				}
-				
-				this._isBusy = false;
-			break;
-		}
-	},
-	
-	_updateBadge: function(containerID) {
-		if (!this._containerData[containerID].likes && !this._containerData[containerID].dislikes) {
-			this._containerData[containerID].badge.hide();
-		}
-		else {
-			this._containerData[containerID].badge.show();
-			
-			// update like counter
-			var $cumulativeLikes = this._containerData[containerID].likes - this._containerData[containerID].dislikes;
-			var $badge = this._containerData[containerID].badge;
-			$badge.removeClass('green red');
-			if ($cumulativeLikes > 0) {
-				$badge.text('+' + WCF.String.formatNumeric($cumulativeLikes));
-				$badge.addClass('green');
-			}
-			else if ($cumulativeLikes < 0) {
-				$badge.text(WCF.String.formatNumeric($cumulativeLikes));
-				$badge.addClass('red');
-			}
-			else {
-				$badge.text('\u00B10');
-			}
-			
-			// update tooltip
-			var $likes = this._containerData[containerID].likes;
-			var $dislikes = this._containerData[containerID].dislikes;
-			$badge.attr('data-tooltip', WCF.Language.get('wcf.like.tooltip', { likes: $likes, dislikes: $dislikes }));
-		}
-	},
-	
-	_updateSummary: function(containerID) {
-		if (!this._containerData[containerID].likes) {
-			this._containerData[containerID].summary.hide();
-		}
-		else {
-			this._containerData[containerID].summary.show();
-			
-			var $users = this._containerData[containerID].users;
-			var $userArray = [];
-			for (var $userID in $users) $userArray.push($users[$userID].username);
-			var $others = this._containerData[containerID].likes - $userArray.length;
-			
-			this._containerData[containerID].summary.children('span').html(WCF.Language.get('wcf.like.summary', { users: $userArray, others: $others }));
-		}
-	},
-	
-	/**
-	 * Sets button active state.
-	 * 
-	 * @param	jquery		likeButton
-	 * @param	jquery		dislikeButton
-	 * @param	integer		likeStatus
-	 */
-	_setActiveState: function(likeButton, dislikeButton, likeStatus) {
-		likeButton.removeClass('active');
-		dislikeButton.removeClass('active');
-		
-		if (likeStatus == 1) {
-			likeButton.addClass('active');
-		}
-		else if (likeStatus == -1) {
-			dislikeButton.addClass('active');
-		}
-	}
+	init: function() {
+		throw new Error("The `WCF.Like` API is obsolete and therefore no longer supported. Please use the current API `WoltLabSuite/Core/Ui/Reaction/Handler` instead.");
```

Comment:
I wonder is this file can be fully removed, not that it is completely non-functional.

Reply:
The file should remain, it provides a sane error handling for horribly outdated implementations. The basic signature is unchanged, eventually preventing implementation errors that would over-shadow the error message above.